beeai/supervisor/main.py

```diff
@@ -0,0 +1,189 @@
+        result = await run_errata_workflow(erratum)
+        if result.reschedule_in >= 0:
+            await queue.schedule_tasks([task], delay=result.reschedule_in)
+        else:
+            await queue.remove_tasks([task])
+
+        logger.info(
+            "Errata %s (%s) processed, status=%s, reschedule_in=%s",
+            erratum.url,
+            erratum.full_advisory,
+            result.status,
+            result.reschedule_in if result.reschedule_in >= 0 else "never",
+        )
+    else:
+        logger.warning("Unknown task type: %s", task)
+
+
+async def do_execute(repeat: bool):
+    async with task_queue(os.environ["REDIS_URL"]) as queue:
+        while repeat:
+            try:
+                await execute_once(queue)
+            except Exception:
+                logger.exception("Error while executing task")
+                await asyncio.sleep(60)
+        else:
+            await execute_once(queue)
+
+
+@app.command()
+def execute(repeat: bool = typer.Option(True)):
+    asyncio.run(do_execute(repeat))
+
+
+async def do_process_issue(key: str):
+    issue = get_issue(key)
+    result = await run_issue_workflow(issue)
+    logger.info(
+        "Issue %s processed, status=%s, reschedule_in=%s",
+        key,
+        result.status,
+        result.reschedule_in if result.reschedule_in > 0 else "never",
+    )
+
+
+@app.command()
+def process_issue(key_or_url: str):
+    if key_or_url.startswith("http"):
+        m = re.match(r"https://issues.redhat.com/browse/([^/?]+)(?:\?.*)?$", key_or_url)
+        if m is None:
+            raise typer.BadParameter(f"Invalid issue URL {key_or_url}")
+        key = m.group(1)
+    else:
+        key = key_or_url
+
+    if not key.startswith("RHEL-"):
+        raise typer.BadParameter("Issue must be in the RHEL project")
+
+    asyncio.run(do_process_issue(key))
+
+
+async def do_process_erratum(id: str):
+    erratum = get_errata_info(id)
+    result = await run_errata_workflow(erratum)
+
+    logger.info(
+        "Errata %s (%s) processed, status=%s, reschedule_in=%s",
+        erratum.url,
+        erratum.full_advisory,
+        result.status,
+        result.reschedule_in if result.reschedule_in >= 0 else "never",
+    )
+
+
+@app.command()
+def process_erratum(id_or_url: str):
+    if id_or_url.startswith("http"):
+        m = re.match(
+            r"https://errata.engineering.redhat.com/advisory/(\d+)$", id_or_url
+        )
+        if m is None:
+            raise typer.BadParameter(f"Invalid advisory URL {id_or_url}")
+        id = m.group(1)
+    else:
+        id = id_or_url
+
+    asyncio.run(do_process_erratum(id))
+
+
+@app.callback()
+def main(debug: bool = False):
+    if debug:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    collector_endpoint = os.environ.get("COLLECTOR_ENDPOINT")
+    if collector_endpoint is not None:
+        setup_observability(collector_endpoint)
```

Comment:
it's a little unfortunate we have two kinds of collectors here... i wonder if it would make sense to call the main on a "scanner" or "scheduler" or something to avoid the clash.